### DISCUSSION TITLE
Extra queries request triggering when using createInsightsMiddleware with Instant Search

### DISCUSSION THREAD

**Discussion by Ali-hd** [algolia/instantsearch#6174] [Q&A] [answered]
Hi, 

I am facing an issue that is causing an extra network request for queries to occur when navigating out of the instant search page.

I have created a sandbox link to check.
https://codesandbox.io/p/devbox/admiring-lake-yshjgk

Open the network tab and observe the queries request and Click on the link "Visit Product Page" on the top left of the website

you will notice an additional queries request being made.

**Comment by Ali-hd** [comment]
Not sure if I should open an issue for this

**Comment by Haroenv** [comment, marked as answer]
This behaviour is happening because there are two insights middleware, and the cleanup of those triggers another search before the other widgets have been removed. 

While this is indeed a bug, I'm not sure when you'd need this specific code in real life. I recommend you either put the `insights` option to false or (recommended) not use the extra manual InsightsMiddleware. If either of those things are done, then no extra queries get sent anymore in my testing.

**Reply by Ali-hd** [reply]
Thank you for answering, 
I didn't realize I can have insights prop as false while keeping the middleware working

**Reply by Haroenv** [reply]
The `insights` parameter essentially just adds an insights middleware, so there's no reason to have both a custom middleware and arguments to the insights prop
